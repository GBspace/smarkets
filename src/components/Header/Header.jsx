import React from 'react';
import { Link } from 'react-router-dom';
import { Typography } from '@material-ui/core';
import {StyledHeader} from './Header.styled';

const Header = () => {
  return (
    <StyledHeader>
      <Typography component="h3" variant="h6">
          <Link className="header-link" to="/">Popular Events</Link>
      </Typography>
    </StyledHeader>
  )
}

export default Header;