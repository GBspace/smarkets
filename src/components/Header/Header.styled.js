import { styled } from "@mui/system";

export const StyledHeader = styled('div')(({theme}) => ({
    a: {
      textDecoration: 'none',
      color: theme.palette.primary.contrastText,
      padding: '5px',
    },
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    padding: '10 0px',
    border: '5px',
    width: '100%',
    backgroundColor: theme.palette.primary.main,
  }));