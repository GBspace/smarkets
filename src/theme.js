import {createTheme} from '@mui/material/styles';

// A custom theme for this app
const customTheme = createTheme({
    palette: {
      type: 'light',
      primary: {
        main: '#696969',
        contrastText: '#282c34'
      },
    },
    overrides: {
      MuiPaper: {
        root: {
          padding: '20px 10px',
          margin: '10px',
          backgroundColor: '#fff',
        },
      },
      MuiButton: {
        root: {
          margin: '5px',
        },
      },
    },
  });
  export {customTheme};