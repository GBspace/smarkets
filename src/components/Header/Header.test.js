import { shallow } from 'enzyme';
import Header from './Header';
import * as React from "react";
import {customTheme} from '../../theme';
import toJson from 'enzyme-to-json';

describe('Header Component', ()=>{
    it('renders correctly', () => {
        const wrapper = shallow(<Header theme={customTheme}/>);
        expect(toJson(wrapper)).toMatchSnapshot();
      });

    it('should display `Popular Events` Link', ()=>{
        const wrapper = shallow(<Header/>);
        const link = wrapper.find('Link');
        expect(link.text()).toBe('Popular Events');
    });
    it('should render Link to homepage', ()=>{
        const wrapper = shallow(<Header/>);
        const link = wrapper.find('Link');
        expect(link.at(0).props().to).toBe('/');
    });
})
