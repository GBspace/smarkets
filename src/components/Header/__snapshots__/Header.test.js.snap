// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component renders correctly 1`] = `
<Styled(div)>
  <WithStyles(ForwardRef(Typography))
    component="h2"
    variant="h5"
  >
    <Link
      className="header-link"
      to="/"
    >
      Popular Events
    </Link>
  </WithStyles(ForwardRef(Typography))>
</Styled(div)>
`;
